deployer/deployer.js

```diff
@@ -236,6 +242,38 @@ class Deployer {
 		return this.wallet.estimateGas(transaction);
 	}
 
+	/**
+	 * 
+	 * Link a library or number of libraries to a contract
+	 * 
+	 * @param {*} libraries The libraries which will be linked to the contract
+	 * @param {*} bytecode The contract's bytecode which be used for linking
+	 */
+	async _prepareBytecode(libraries, bytecode) {
+		if (isValidLibrary(libraries)) {
+			return await this._linkLibrary(libraries, bytecode);
+		}
+
+		return bytecode;
+	}
+
+	async _linkLibrary(libraries, bytecode) {
+		if (bytecode === '' || bytecode === undefined) {
+			throw new Error(`"${bytecode}" is not a valid bytecode.`);
+		}
+
+		for (const libraryName in libraries) {
+			if (libraries.hasOwnProperty(libraryName)) {
+				const libraryAddress = libraries[libraryName];
+
+				var regex = new RegExp("__" + libraryName + "_+", "g");
```

Comment:
This seems like very complex custom logic that should not be in this class. Please put this in util. You might make a new linking util class. If you do so move some of the contract util logic there too.